apps/server/lib/mix/tasks/package.ex

```diff
@@ -0,0 +1,311 @@
+  won't realize that the correct apps are installed. It uses two different VMs, so making an `escript`
+  will fail, as the second VM needs to find lexical's modules _somewhere. Releases seem ideal, and we
+  used them for a while, but they need to match the _exact_ Elixir and Erlang versions they were compiled on,
+  right down to the patch level. This is much, much too strict for a project that needs to be able to run
+  on a variety of elixir / erlang versions.
+
+  An ideal packaging system will have the following properties:
+
+    * It creates a self contained artifact directory
+    * It will run under a variety of versions of Elixir and Erlang
+    * It allows namespaced apps to run
+    * It allows multiple VMs to find elixir's modules
+    * It allows us to package non-elixir resources (mainly launcher scripts) and access them during runtime
+
+  This packaging system meets all of the above parameters.
+  It works by examining the server app's dependencies, namespacing what's required and then packaging them
+  in to [Erlang archive files](https://www.erlang.org/doc/man/code.html#loading-of-code-from-archive-files)
+  in a `lib` directory. Similarly, the configuration is copied to a `config` directory, and the launcher
+  scripts are copied to a `bin` directory.
+
+  The end result is a release-like filesystem, but without a lot of the erlang booting stuff. Bootstrapping
+  accomplished by simple scripts that reside in the project's `/bin` directory, and the `Lexical.Server.Boot`
+  module, which loads applications and their modules.
+
+  ## Command line options
+
+  * `--path` - The package will be written to the path given. Defaults to `./build/dev/lexical`. If the
+    `--zip` option is specified, The name of the zip file is determined by the last entry of the path.
+    For example, if the `--path` option is `_build/dev/output`, then the name of the zip file will be
+    `output.zip`.
+  * `--zip`  - The resulting package will be zipped. The zip file will be placed in the current directory,
+    and the package directory will be deleted
+
+
+  ## Directory structure
+  ```text
+  bin/
+    start_lexical.sh
+    debug_shell.sh
+  lib/
+    lx_common.sh
+    lx_remote_control.sh
+    lx_server.ez
+    ...
+  config/
+    config.exs
+    dev.exs
+    prod.exs
+    test.exs
+    runtime.exs
+  priv/
+    port_wrapper.sh
+    ...
+  consolidated/
+    Elixir.(consolidated protocol module).beam
+  ```
+
+  On boot, the `ERL_LIBS` environment variable is set to the `lib` directory so all of the `.ez` files are
+  picked up by the code server. Similarly, the `config`, `consolidated` and `priv` directories are added
+  to the code search path with the `-pa` argument.
+  """
+
+  alias Mix.Tasks.Namespace
+
+  @options [
+    strict: [
+      path: :string,
+      zip: :boolean
+    ]
+  ]
+
+  @execute_permisson 0o755
+
+  def run(args) do
+    Mix.Task.run(:compile)
+    {opts, _, _} = OptionParser.parse(args, @options)
```

Comment:
Didn't know there's an `OptionParser` module in the standard library. Pretty neat!

Reply:
Yeah, it's great. There's also a version comparator. Love it.